pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -437,8 +437,14 @@ func (rq *ResourceQuotaController) Sync(discoveryFunc NamespacedResourcesFunc, p
 			utilruntime.HandleError(fmt.Errorf("failed to sync resource monitors: %v", err))
 			return
 		}
-		if rq.quotaMonitor != nil && !controller.WaitForCacheSync("resource quota", stopCh, rq.quotaMonitor.IsSynced) {
+		// wait for caches to fill for a while (our sync period).
+		// this protects us from deadlocks where available resources changed and one of our informer caches will never fill.
+		// informers keep attempting to sync in the background, so retrying doesn't interrupt them.
+		// the call to resyncMonitors on the reattempt will no-op for resources that still exist.
+		// note that workers stay paused until we successfully resync.
```

Comment:
when we pause the workers, what's the actual implication for everyone above?  will admission block?

Reply:
this is the controller, reconciliation would pause. this comment is not actually correct, it's left from the copy from the GC controller, where we do stay paused for safety